From 1e08d65c353b23c037b6119ebd91822bd052d8c0 Mon Sep 17 00:00:00 2001
From: Abhi Teja Cholleti <achollet@qti.qualcomm.com>
Date: Fri, 1 Aug 2025 13:18:55 +0530
Subject: [PATCH 1/1] Add multicast to unicast and vendor elements support

Signed-off-by: Abhi Teja Cholleti <achollet@qti.qualcomm.com>
---
 src/nl80211/wld_hostapd_cfgFile.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/nl80211/wld_hostapd_cfgFile.c b/src/nl80211/wld_hostapd_cfgFile.c
index e89001d..914497e 100644
--- a/src/nl80211/wld_hostapd_cfgFile.c
+++ b/src/nl80211/wld_hostapd_cfgFile.c
@@ -775,6 +775,12 @@ static bool s_setVapCommonConfig(T_AccessPoint* pAP, swl_mapChar_t* vapConfigMap
     swl_mapCharFmt_addValInt32(vapConfigMap, "ppe_vp", pRad->ppevpType);
     SAH_TRACEZ_INFO(ME, "%s: PPEVP set ppe_vp = %u", pRad->Name, pRad->ppevpType);
 
+    /* Set multicast_to_unicast = 1 */
+    swl_mapCharFmt_addValInt32(vapConfigMap, "multicast_to_unicast", 1);
+
+    /* Set vendor_elements if fixed */
+    swl_mapCharFmt_addValStr(vapConfigMap, "vendor_elements", "%s", "dd050026861401");
+
     swl_mapCharFmt_addValInt32(vapConfigMap, "ap_isolate", pAP->clientIsolationEnable);
     swl_mapChar_add(vapConfigMap, "ignore_broadcast_ssid", pAP->SSIDAdvertisementEnabled ? "0" : "2");
     if(!pAP->enable || !pRad->enable) {
-- 
2.34.1

