From 025f95a23fddce5ead4642af2eb88d839ea38175 Mon Sep 17 00:00:00 2001
From: VedhaLoganathan <vedhaval@qti.qualcomm.com>
Date: Tue, 5 May 2026 16:17:14 +0530
Subject: [PATCH] ML Reconfiguration Link Removal on Vaps

Change-Id: I3321fbf46348a1cf63b1a7a5a47ea748e3ec046a
Signed-off-by: VedhaLoganathan <vedhaval@qti.qualcomm.com>
---

--- a/src/Plugin/wifiGen_rad.c
+++ b/src/Plugin/wifiGen_rad.c
@@ -602,7 +602,7 @@ int wifiGen_rad_supports(T_Radio* pRad,
     rc = wld_rad_nl80211_getInterfaceInfo(pRad, &ifaceInfo);
     ASSERT_FALSE(rc < SWL_RC_OK, rc, ME, "Fail to get nl80211 rad iface info");
     pRad->wiphy = ifaceInfo.wiphy;
-    swl_str_copy(pRad->wiphyName, sizeof(pRad->wiphyName), wiphyInfo.name);
+    swl_str_copy(pRad->wiphyName, sizeof(pRad->wiphyName), pRad->Name);
     pRad->wDevId = ifaceInfo.wDevId;
     wifiGen_setRadEvtHandlers(pRad);
 
