From d3c21c34e30dcffd2576f1b8f01f00eda6c9d1ff Mon Sep 17 00:00:00 2001
From: VedhaLoganathan <vedhaval@qti.qualcomm.com>
Date: Sun, 24 May 2026 01:20:05 +0530
Subject: [PATCH] Update APMLD DM on ML Reconfiguration

Signed-off-by: VedhaLoganathan <vedhaval@qti.qualcomm.com>
---

diff --git a/src/Plugin/wifiGen_fsm.c b/src/Plugin/wifiGen_fsm.c
index f4d4bf8..10f6592 100644
--- a/src/Plugin/wifiGen_fsm.c
+++ b/src/Plugin/wifiGen_fsm.c
@@ -906,7 +906,7 @@ static bool s_doDisableHostapd(T_Radio* pRad) {
      */
     wifiGen_hapd_deauthKnownStations(pRad, true);
     if(wld_rad_hostapd_disable(pRad) >= 0) {
-        //wld_rad_resetApMldLinks(pRad);
+        wld_rad_resetApMldLinks(pRad);
     }
     pRad->fsmRad.timeout_msec = 500;
     return true;
diff --git a/src/wld_apMld.c b/src/wld_apMld.c
index 60e5053..a36224d 100644
--- a/src/wld_apMld.c
+++ b/src/wld_apMld.c
@@ -390,6 +390,21 @@ static swl_rc_ne s_updateDmParams(wld_mld_t* pMld) {
     swl_macChar_t mldMacAddress;
     memset(&mldMacAddress, 0, sizeof(mldMacAddress));
 
+    uint32_t nMldLinks = 0;
+    amxc_llist_for_each(it, &pMld->links) {
+        wld_mldLink_t* pLink = amxc_container_of(it, wld_mldLink_t, it);
+        T_SSID* pLinkSSID = wld_mld_getLinkSsid(pLink);
+        if(pLinkSSID && (pLinkSSID->mldUnit == pMld->unit)) {
+            nMldLinks++;
+        }
+    }
+
+    if(nMldLinks == 0) {
+        SAH_TRACEZ_WARNING(ME, "MLD unit %d has no links - flushing", pMld->unit);
+        s_flushMldParams(pMld);
+        return SWL_RC_OK;
+    }
+
     if(pMld->pPrimLink != NULL) {
         T_SSID* pPrimSSID = wld_mld_getLinkSsid(pMld->pPrimLink);
         if((pPrimSSID != NULL) && !swl_mac_binIsNull((swl_macBin_t*) pPrimSSID->MACAddress) && wld_ssid_hasValidMLDLinkID(pPrimSSID)) {
