From abb651de039129901453bf7d6ca5786ff5549580 Mon Sep 17 00:00:00 2001
From: Nikhil Gurram <nikhgurr@qti.qualcomm.com>
Date: Mon, 25 May 2026 10:55:10 +0530
Subject: [PATCH 1/1] change

Signed-off-by: Nikhil Gurram <nikhgurr@qti.qualcomm.com>
---
 src/core/database.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/core/database.c
+++ b/src/core/database.c
@@ -965,7 +965,7 @@ int OpenUspDatabase(char *db_file)
     #define SQLITE_MINIMUM_PAGE_SIZE 512
     if (st.st_size < SQLITE_MINIMUM_PAGE_SIZE)
     {
-        USP_LOG_Error("%s: USP database (%s) size (%ld) is less than the sqlite minimum page size (%d)", __FUNCTION__, db_file, (long)st.st_size, SQLITE_MINIMUM_PAGE_SIZE);
+        USP_LOG_Error("%s: USP database (%s) size (%lld) is less than the sqlite minimum page size (%d)", __FUNCTION__, db_file, (long long)st.st_size, SQLITE_MINIMUM_PAGE_SIZE);
         err = ReInitializeDatabase(db_file, true);
         goto exit;
     }
