From 935b6d076abcf977d75c634aed14ffef72ad72d1 Mon Sep 17 00:00:00 2001
From: Mohan Raj <mrajraje@qti.qualcomm.com>
Date: Mon, 23 Jun 2025 20:13:29 +0530
Subject: [PATCH] qca: Fix_ODL_generator_producing_invalid_wld.odl

Change-Id: Ibf065268899b0c9a8c072da7ad42583103cc651b
Signed-off-by: Mohan Raj <mrajraje@qti.qualcomm.com>

diff --git a/files/netlayout_from_uci.sh b/files/netlayout_from_uci.sh
index 1b7e73d..bebc82b 100755
--- a/files/netlayout_from_uci.sh
+++ b/files/netlayout_from_uci.sh
@@ -385,6 +385,11 @@ wait_for_uci_wireless() {
 	phy_count=$(iw phy | grep -c Wiphy)
 	uci_count=$(count_matching_lines 'wifi-device.*radio' /etc/config/wireless)
 
+	while [ $uci_count -lt 3 ]; do
+		uci_count=$(count_matching_lines 'wifi-device.*radio' /etc/config/wireless)
+		sleep 1
+	done
+
 	while ([ "$phy_count" -lt "$min_phy" ] || [ "$phy_count" -ne "$uci_count" ]) && [ $elapsed -lt $timeout ]; do
 		phy_count=$(iw phy | grep -c Wiphy)
 		uci_count=$(count_matching_lines 'wifi-device.*radio' /etc/config/wireless)
