From d7a822c4d186e024695843476bc678293fee3a35 Mon Sep 17 00:00:00 2001
From: Nikhil Gurram <nikhgurr@qti.qualcomm.com>
Date: Fri, 24 Oct 2025 19:57:53 +0530
Subject: [PATCH 1/1] commit

Signed-off-by: Nikhil Gurram <nikhgurr@qti.qualcomm.com>
---
files/common.netlayout_from_uci.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/files/common.netlayout_from_uci.sh b/files/common.netlayout_from_uci.sh
index 1b7e73d..2691578 100755
--- a/files/common.netlayout_from_uci.sh
+++ b/files/common.netlayout_from_uci.sh
@@ -378,6 +378,7 @@ wait_for_uci_wireless() {
 	local min_phy=0
 	local timeout_uci=$(uci -q get wireless.vendor.phy_detect_timeout)
 	local min_phy_uci=$(uci -q get wireless.vendor.min_phy)
+	local pci_count=$(ls /sys/kernel/debug/ath12k/ |wc -l)
 
 	[ -n "$timeout_uci" ] && timeout=$timeout_uci
 	[ -n "$min_phy_uci" ] && min_phy=$min_phy_uci
@@ -385,6 +386,11 @@ wait_for_uci_wireless() {
 	phy_count=$(iw phy | grep -c Wiphy)
 	uci_count=$(count_matching_lines 'wifi-device.*radio' /etc/config/wireless)
 
+	while [ $uci_count -lt $pci_count ]; do
+		uci_count=$(count_matching_lines 'wifi-device.*radio' /etc/config/wireless)
+		sleep 1
+	done
+
 	while ([ "$phy_count" -lt "$min_phy" ] || [ "$phy_count" -ne "$uci_count" ]) && [ $elapsed -lt $timeout ]; do
 		phy_count=$(iw phy | grep -c Wiphy)
 		uci_count=$(count_matching_lines 'wifi-device.*radio' /etc/config/wireless)
-- 
2.34.1

